pinot-common/src/main/java/org/apache/pinot/common/utils/DataSchema.java

```diff
@@ -490,12 +540,24 @@ public Object toInternal(Object value) {
           return ((boolean) value) ? 1 : 0;
         case TIMESTAMP:
           return ((Timestamp) value).getTime();
+        case TIMESTAMP_NTZ:
+          return ((LocalDateTime) value).atZone(ZoneId.of("UTC")).toInstant().toEpochMilli();
```

Comment:
@bziobrowski IIRC you knew a way to do this without allocating so many objects. Am I right?

Reply:
In this specific case, this can be rewritten with (AFAIK) zero allocation with:

```suggestion
          return ((LocalDateTime) value).toEpochSecond(ZoneOffset.UTC) * 1000;
```

Reply:
> In this specific case, this can be rewritten with (AFAIK) zero allocation with:

Thanks! But I think this will lose millisecond accuracy😂

Reply:
@gortiz I think that snippet isn't correct because code assumes offset is constant (and actually 0 while local is likely not :) ) 
@chunxiaozheng  Please consider : 
```java
  public static void main(String[] args) {
    LocalDateTime dt = LocalDateTime.of(2024, 11, 8, 12, 13, 14, 231000000);

    ZoneId zoneId = ZoneId.of("Europe/Warsaw");// zone should be cached
    ZoneRules rules = zoneId.getRules();
    ZoneOffset offset = rules.getOffset(dt);

    System.out.println("to epoch (UTC) " + dt.toEpochSecond(ZoneOffset.UTC) * 1000);
    System.out.println("to epoch " + dt.toEpochSecond(offset) * 1000);
    System.out.println("to epoch plus nanos " + dt.toEpochSecond(offset) * 1000 + dt.getNano() / 1000000);
  }
```

Reply:
> @gortiz I think that snippet isn't correct because code assumes offset is constant (and actually 0 while local is likely not :) ) @chunxiaozheng Please consider :
> 
> ```java
>   public static void main(String[] args) {
>     LocalDateTime dt = LocalDateTime.of(2024, 11, 8, 12, 13, 14, 231000000);
> 
>     ZoneId zoneId = ZoneId.of("Europe/Warsaw");// zone should be cached
>     ZoneRules rules = zoneId.getRules();
>     ZoneOffset offset = rules.getOffset(dt);
> 
>     System.out.println("to epoch (UTC) " + dt.toEpochSecond(ZoneOffset.UTC) * 1000);
>     System.out.println("to epoch " + dt.toEpochSecond(offset) * 1000);
>     System.out.println("to epoch plus nanos " + dt.toEpochSecond(offset) * 1000 + dt.getNano() / 1000000);
>   }
> ```

Thanks! The implementation of TIMESTAMP type without time zone, such as `TIMESTAMP` in Flink and `TIMESTAMP_NTZ` type in Spark, will parse according to UTC time to ensure that the time queried in any time zone is consistent. Therefore, for TIMESTAMP types without time zones, I believe it is need to use a constant(UTC time zone) for parsing.